libs/game/sprite.ts

```diff
@@ -71,27 +101,31 @@ class Sprite extends sprites.BaseSprite {
     _ay: Fx8
     _fx: Fx8 // friction
     _fy: Fx8 // friction
+    _sx: Fx8 // scale
+    _sy: Fx8 // scale
+    _width: Fx8 // scaled width
+    _height: Fx8 // scaled height
 
     //% group="Physics" blockSetVariable="mySprite"
     //% blockCombine block="x" callInDebugger
     get x(): number {
-        return Fx.toFloat(this._x) + (this._image.width / 2)
+        return Fx.toFloat(this._x) + (this.width / 2)
```

Comment:
we might wanna just do these calculations with fx8 since we have _width as fx8 now but doesn't matter too much